src/service/ProductManagementService.js

```diff
@@ -0,0 +1,162 @@
+import { CONFIRMATION_RESPONSES } from '../constants/constants.js';
+import getUserConfirmation from '../utils/getUserConfirmation.js';
+
+class ProductManagementService {
+  #stock;
+  #promotion;
+  #productsInfo;
+
+  constructor(stock, promotion) {
+    this.#stock = stock;
+    this.#promotion = promotion;
+    this.#productsInfo = [];
+  }
+
+  // 구매할 상품 및 수량 저장
+  initProductsInfo(productsInfo) {
+    this.#productsInfo = productsInfo;
+  }
+
+  // 구매할 상품 및 수량 전체 정보 반환
+  getProductsInfo() {
+    return this.#productsInfo;
+  }
+
+  // 재고 수량 초과하는지 확인
+  hasSufficientStock(productName, requestedQuantity) {
+    const totalQuantity = this.#calculateTotalQuantity(productName);
+    return totalQuantity >= requestedQuantity;
+  }
+
+  // 구매할 상품들에 대한 처리 로직
+  async processProducts() {
+    for (const { name, quantity } of this.#productsInfo) {
+      if (this.#isGeneralProduct(name, quantity)) continue;
```

Comment:
`isGeneralProduct`는 프로모션 미적용 상품인지 확인하는 것 뿐만 아니라 프로모션 미적용 상품의 결제 프로세스도 진행되는 것으로 보입니다. 따라서 `isGeneralProduct`의 메서드 명을 `processGeneralProduct`로 수정하고 아래와 같이 로직을 수정하면 어떨까요??

```suggestion
      if (!this.#isAvailablePromotion(name)) {
        this.#processGeneralProduct(name, quantity);
        continue;
      }
```

```js
  #processGeneralProduct(name, quantity) {
    this.#stock.updateGeneralStockInfo(name, quantity);
    this.#initGiftCount(name);
  }
```

Reply:
원래는 지예님 말씀해주신 방식대로 구현을 했었는데요, `indent depth` 가 `3`이 되어버려서 `2`로 줄이려다 보니 리팩토링 하는 과정에서 코드가 지금과 같은 형태로 변형되어 버렸습니다 🥲 그래서 `retry` 브랜치 보시면 이번에 두 번째로 다시 구현해본 코드인데, 여기서는 해당 로직을 구현할 때 이런 괴상한 선언형 로직이 사라졌습니다 ! ㅎㅎ